be/main.py

```diff
@@ -78,38 +69,80 @@ async def mavedb_search(params: Annotated[SearchParams, Query()]) -> MaveDBRespo
         }
     }, "aggs": defaultdict(dict)}
 
-    # Adding aggregation fields
-    for aggregation_field in AGGREGATION_FIELDS:
-        search_body["aggs"][aggregation_field] = {
-            "terms": {"field": aggregation_field}
-        }
+    # Adding aggregation fields.
+    if aggregation_fields:
+        for aggregation_field in aggregation_fields:
+            print(aggregation_field)
+            search_body["aggs"][aggregation_field] = {
+                "terms": {"field": aggregation_field}
+            }
 
     # Adding sort field and sort order
     search_body['sort'] = [{params.sort_field: {"order": params.sort_order}}]
 
+    # Performing the search.
     try:
         # Execute the async search request.
-        response = await app.state.es_client.search(index="mavedb", body=search_body)
+        response = await app.state.es_client.search(index=index_name, body=search_body)
         # Extract total count and hits.
         total = response["hits"]["total"]["value"]
         hits = [r["_source"] for r in response["hits"]["hits"]]
-        # Return the response with pagination info.
-        return MaveDBResponse(total=total, start=params.start, size=params.size,
-                              results=hits,
-                              aggregations=response["aggregations"])
+        aggregations = response["aggregations"]
+        return total, hits, aggregations
+
     except Exception as e:
         # Handle Elasticsearch errors.
         raise HTTPException(status_code=500, detail=f"Search error: {str(e)}")
 
-
-@app.get("/mavedb/search/{record_id}")
-async def mavedb_details(record_id: Annotated[
-    str, Path(description="Record id")]) -> MaveDBDetailsResponse:
+async def elastic_details(index_name, record_id):
     try:
-        response = await app.state.es_client.search(index="mavedb",
-                                   q=f"_id:{urllib.parse.quote(record_id)}")
+        response = await app.state.es_client.search(
+            index=index_name,
+            q=f"_id:{urllib.parse.quote(record_id)}"
+        )
         hits = [r["_source"] for r in response["hits"]["hits"]]
-        return MaveDBDetailsResponse(results=hits)
+        return hits
     except Exception as e:
         # Handle Elasticsearch errors.
         raise HTTPException(status_code=500, detail=f"Search error: {str(e)}")
+
+
+# MaveDB.
+
+@app.get("/mavedb/search")
+async def mavedb_search(params: Annotated[MaveDBSearchParams, Query()]) -> ElasticResponse[MaveDBData, MaveDBAggregationResponse]:
```

Comment:
This line seems to be more than 80

Reply:
Thank you, fixed in https://github.com/EMBL-EBI-ABC/PerturbationCatalogue/pull/90/commits/f2ae7cc00e5e5d2bb34ea6dff070e749afffbd12